### ISSUE TITLE
CVMFS_STREAMING_CACHE=yes causes authorization errors and file corruption

### ISSUE THREAD

**Issue by DrDaveD** [cvmfs/cvmfs#3629]
In cvmfs-contrib/config-repo#266 we set `CVMFS_STREAMING_CACHE=yes` for all osgstorage.org and storage.igwn.org repositories when using the config-osg configuration repository.  Unfortunately, LIGO started noticing file corruption as reported in this [osg ticket](https://support.opensciencegrid.org/support/tickets/public/3b2c34dc8b2833f1a081e86e57c4fa0c0259f52baf686ba727bf3446c6df8848).

The first problem is that simply enabling this option causes authentication errors on all the authenticated storage.igwn.org repositories, for example ligo.storage.igwn.org.  I can provide a full debuglog trace if that's desired, but here's a subset of what all the connection attempts look like:
```
(curl) {header/out} GET //igwn/ligo/frames/O4/hoft_C00/H1/H-H1_HOFT_C00-140/H-H1_HOFT_C00-1401489994-2.gwf HTTP/1.1
Host: sc-cache.chtc.wisc.edu:8443
Accept: */*
Connection: Keep-Alive
User-Agent: cvmfs Fuse 2.11.3
cvmfs-info: /igwn/ligo/frames/O4/hoft_C00/H1/H-H1_HOFT_C00-140/H-H1_HOFT_C00-1401489994-2.gwf    [07-08-2024 13:48:33 CDT]
(curl) {ssldata/in} <snip>    [07-08-2024 13:48:33 CDT]
(curl) {header/in} HTTP/1.1 403 Forbidden    [07-08-2024 13:48:33 CDT]
(download) http status error code: HTTP/1.1 403 Forbidden [403]    [07-08-2024 13:48:33 CDT]
```

More concerning, despite the fact that all the external hosts are causing errors, corrupt data is provided to the user and the user is not given an error.  Here's how to reproduce that problem on a non-authenticated repository:
1.  First, make a copy of the test file to compare against with `cp /cvmfs/des.osgstorage.org/stash/kherner/load_testing/testfile.yujun /tmp/testfile.yujun.good`
2. Set `CVMFS_STREAMING_CACHE=yes` in /etc/cvmfs/config.d/des.osgstorage.org.local.
3. Find a port on localhost that returns Connection refused.  I am using port 3128 for an example.  Set it in the external host list twice, either with `CVMFS_EXTERNAL_URL="http://localhost:3128;http://localhost:3128"` (and reloading/remounting) or by mounting `/cvmfs/des.osgstorage.org` and running `sudo cvmfs_talk -i des.osgstorage.org external host set "http://localhost:3128;http://localhost:3128"`.
4. Make sure cache is cleared and new option takes effect, for example with `sudo cvmfs_config reload -c des.osgstorage.org`
5. `cp /cvmfs/des.osgstorage.org/stash/kherner/load_testing/testfile.yujun /tmp/testfile.yujun` and see that there is no error but the content does not match the previously copied file.  In fact, if you reload or remount, each time you copy the file it gets a different contents.

**Comment by jblomer** [comment]
Thank you for the report!  I can reproduce the file corruption, more soon.

**Comment by jblomer** [comment]
The corruption should be fixed by #3630. A plain error in the handling of network I/O errors, which unfortunately we did not properly test for.

I'll continue to look into the rest of this issue.
